### PR TITLE
Updated README.md

### PR DESCRIPTION
**Pull request by ramya25**
## Summary
Included step by step process to run this sample.

## Issue Type
- Bug fixing

## Starter Names
  - active directory spring boot starter

## Additional Information
- Updating the README.md for azure active directory spring boot sample

### CODE REVIEW

File: azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md [hunk 1]
```diff
@@ -1,25 +1,112 @@
...
+ `azure.activedirectory.client-secret=ABCDEFGHIJKLMNOOPQRSTUVWXYZABCDEFGHIJKLMNOPQ`
+1. List all the AAD groups `ActiveDirectoryGroups` that you want to have a Spring Security role object mapping to it. The role objects can then be used to manage access to resources that is behind Spring Security. e.g.
+ `azure.activedirectory.active-directory-groups=group1,group2` (groups that you created in your Azure AD tenant)
+ 
+ ### Step 4: Change Role_group1 to your group
+1. You can use `@PreAuthorize` annotation or `UserPrincipal` to manage access to web API based on user's group membership. You will need to change `ROLE_group1` to groups you want to allow to access the API in `TodoListController.java` or you will get "Access is denied".
+   
+   
+### Step 5: Angular JS
+1. In `app.js`, make following changes. The client leverages Azure AD library for JS to handle AAD authentication in single page application. The following snippet of code configures adal provider for your registered app.
```
Comment (Incarnation-p-lee) [inline azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md]: seems unnecessary to index as there is only one item here.

Comment (ramya25) [reply azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md]: Removed the index.

File: azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md [hunk 2]
```diff
@@ -1,24 +1,112 @@
...
+- A [client secret](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#create-a-new-application-secret) for the registered application.
+- Configure groups in your Azure AD tenant with your users in that groups, see [how to create groups](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal)
+- The sample retrieves user's group membership using Azure AD graph API which requires the registered app to have `Direcory.AccessAsUser.All` "Access the directory as the signed-in user" under `Delegated Permissions`. You need AAD admin privilege to be able to grant the permission in API ACCESS -> Required permission.
+
+#### Note
+* If you are not the admin, you need consent from your admin for the the `Directory.AccessAsUser.All` permission. For details see [Directory Permissions](https://docs.microsoft.com/en-us/graph/permissions-reference#directory-permissions)
```
[the remark this answers is not in the document]
Comment (ramya25) [reply azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md]: changed

File: azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md [hunk 3]
```diff
@@ -1,24 +1,112 @@
...
+
+### Step 2:  Register the sample with your Azure Active Directory tenant
+
+To register it follow the steps below or follow the guide [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-protocols-oauth-code#register-your-application-with-your-ad-tenant).
+
+#### Choose the Azure AD tenant where you want to create your applications
+
+As a first step you'll need to:
+
+1. Sign in to the [Azure portal](https://portal.azure.com) using either a work or school account or a personal Microsoft account.
```
Comment (kalyankrishna1) [inline azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md]: >or a personal Microsoft account [](start = 97, length = 31)

review this

File: azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md [hunk 4]
```diff
@@ -1,24 +1,112 @@
...
+To run this sample, you'll need:
+
+- JDK 1.8 and above
+- [Maven](http://maven.apache.org/) 3.0 and above
+- An Intenet connection
+- A Windows machine (necessary if you want to run the app on Windows)
+- An OS X machine (necessary if you want to run the app on Mac)
+- A Linux machine (necessary if you want to run the app on Linux)
+- An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, see [How to get an Azure AD tenant](https://azure.microsoft.com/en-us/documentation/articles/active-directory-howto-tenant/)
+- A user account in your Azure AD tenant. This sample will not work with a Microsoft account (formerly Windows Live account). Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a Microsoft account and have never created a user account in your directory before, you need to do that now.
```
Comment (kalyankrishna1) [inline azure-spring-boot-samples/azure-active-directory-spring-boot-sample/README.md]: >Microsoft account [](start = 75, length = 17)

Microsoft personal Account

**Pull request merged**